packages/toast/src/toast-manager.tsx

```diff
@@ -143,30 +161,33 @@ export class ToastManager extends React.Component<Props, State> {
       onCloseComplete: options.onCloseComplete,
       onRequestRemove: () => this.removeToast(String(id), position),
       status: options.status,
+      requestClose: false,
     }
   }
 
   /**
-   * Requests to close a toast based on it's id and position
+   * Requests to close a toast based on its id and position
    */
   closeToast = (id: ToastId) => {
     this.setState((prevState) => {
       const position = getToastPosition(prevState, id)
 
-      if (!position) return prevState
+      if (!position) {
+        return prevState
+      }
 
       return {
         ...prevState,
         [position]: prevState[position].map((toast) => ({
           ...toast,
-          requestClose: toast.id == id,
+          requestClose: toast.id == id ? true : toast.requestClose,
```

Comment:
this was leading to the actual bug reported: per position, only one toast at max was closed at a time when using `closeAll` because it was always overwriting the previous value